doc/user_guide/Server_Deployment.md

```diff
@@ -209,6 +209,47 @@ Finally, to deploy a Panel app to App Engine run `gcloud app create` and `gcloud
 
 For detailed information and steps, check out this [example](https://towardsdatascience.com/deploy-a-python-visualization-panel-app-to-google-cloud-cafe558fe787?sk=98a75bd79e98cba241cc6711e6fc5be5) on how to deploy a Panel app to App Engine and this [example](https://towardsdatascience.com/deploy-a-python-visualization-panel-app-to-google-cloud-ii-416e487b44eb?sk=aac35055957ba95641a6947bbb436410) on how to deploy a Panel app to Cloud Run.
 
+#### Hugging Face
+
+The guides below assumes you have already signed up and logged into your account at [huggingface.co](https://huggingface.co/).
+
+#### Duplicate an existing space
+
+The easiest way to get started is to  [search](https://huggingface.co/spaces), find and duplicate an existing space. A simple space is
+[MarcSkovMadsen/awesome-panel](https://huggingface.co/spaces/MarcSkovMadsen/awesome-panel).
+
+- Open the space [MarcSkovMadsen/awesome-panel](https://huggingface.co/spaces/MarcSkovMadsen/awesome-panel).
+- Click the 3 dots and select *Duplicate this Space*.
+
+<img src="../_static/hugging-face-duplicate.png" style="width:67%"></img>
+
+- Follow the instructions to finish the duplication.
+
+Once you have finalized the duplication you will need to take a look at the `app.py` file in the new space to figure out what to replace.
+
+In this example you will need to
+
+- Update the contents of the `panel_app.py` file.
+
+<img src="../_static/hugging-face-app-py.jpg" style="width:67%"></img>
+
+#### Creating a new space from scratch
+
+You can deploy Panel to Hugging Face Spaces as a [*Custom Python Space*](https://huggingface.co/docs/hub/spaces-sdks-python). For a general introduction to Hugging Face Spaces see the [Spaces Overview](https://huggingface.co/docs/hub/spaces-overview).
+
+If you want to create a new space from scratch you will need to use a Gradio space.
+
+<img src="../_static/hugging-face-gradio-space.jpg" style="width:67%"></img>
+
+A Gradio space will serve the app via the commmand `python app.py`. I.e. you cannot run `panel serve app.py`.
+
+To work around this your `app.py` will need to either
+
+- Use `subprocess` to run `panel serve some_app_file.py` or
+- Use `pn.serve` to serve one or more functions.
```

Comment:
@philippjfr . Should we document in full here (i.e. include files with content) or should we keep it to the general guidelines as here?